src/calico_manifests.py

```diff
@@ -326,6 +340,7 @@ def __init__(self, charm, charm_config, etcd: EtcdReactiveRequires, cni_config:
             PatchValuesKubeControllers(self),
             PatchIPAutodetectionMethod(self),
             PatchEtcdPaths(self),
+            PatchCDKOnCAChange(self),
```

Comment:
do you have any concerns about adding `ManifestLabel(self)` to this?

it's a common manipulator designed to add some labels about which charm deployed this thing -- helps when you want to search all resources applied by this charm

```
from ops.manifests import ManifestLabel
```

and add it to this list of manipulations

Reply:
The thing that concerns me the most here is that every resource is going to have the same label, right? So I think when the CIDR is changed, it's going to restart other workloads that we probably wouldn't want to restart.

Reply:
`ManifestLabel` won't apply the `cdk-restart-on-ca-change` label, just these:

```python
            labels = {
                "juju.io/application": self.manifests.model.app.name,
                "juju.io/manifest": self.manifests.name,
                "juju.io/manifest-version": f"{self.manifests.name}-{version}",
            }
```

They're just little identifier flags that let you know where the resource comes from